src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala

```diff
@@ -0,0 +1,87 @@
+package mesosphere.marathon
+package api.akkahttp.v2
+
+import akka.http.scaladsl.server.Route
+import mesosphere.marathon.api.akkahttp.Controller
+import mesosphere.marathon.api.akkahttp.Directives.extractInstanceId
+import mesosphere.marathon.api.akkahttp.PathMatchers.ExistingAppPathId
+import mesosphere.marathon.core.group.GroupManager
+import mesosphere.marathon.core.instance.Instance
+import mesosphere.marathon.plugin.auth.Authenticator
+import mesosphere.marathon.state.PathId
+
+class PodsController(
+    val groupManager: GroupManager)(
+    implicit
+    val authenticator: Authenticator) extends Controller {
+
+  import mesosphere.marathon.api.akkahttp.Directives._
+
+  def capability(): Route =
+    authenticated.apply { implicit identity =>
+      complete("")
+    }
+
+  def create(): Route = ???
+
+  def update(podId: PathId): Route = ???
+
+  def findAll(): Route = ???
+
+  def find(podId: PathId): Route = ???
+
+  def remove(podId: PathId): Route = ???
+
+  def status(podId: PathId): Route = ???
+
+  def versions(podId: PathId): Route = ???
+
+  def version(version: String): Route = ???
+
+  def allStatus(): Route = ???
+
+  def killInstance(instanceId: Instance.Id): Route = ???
+
+  def killInstances(podId: PathId): Route = ???
+
+  override val route: Route =
+    head {
+      capability()
+    } ~
+      get {
+        findAll()
+      } ~
+      post {
+        create()
+      } ~
+      pathPrefix(ExistingAppPathId(groupManager.rootGroup)) { podId =>
```

Comment:
Should we rename the matcher to `ExistingPathId` or `ExistingRunSpecId`?

Reply:
We actually should use another one since `ExistingAppPathId` is not returning 400 if the app id is invalid.